AsyncDisplayKit/ASCellNode.m

```diff
@@ -32,6 +40,35 @@ - (instancetype)init
   return self;
 }
 
+- (instancetype)initWithViewControllerBlock:(ASDisplayNodeViewControllerBlock)viewControllerBlock
+{
+  return [self initWithViewControllerBlock:viewControllerBlock didLoadBlock:nil];
+}
+
+- (instancetype)initWithViewControllerBlock:(ASDisplayNodeViewControllerBlock)viewControllerBlock didLoadBlock:(ASDisplayNodeDidLoadBlock)didLoadBlock
+{
+  if (!(self = [super init]))
+    return nil;
+  
+  ASDisplayNodeAssertNotNil(viewControllerBlock, @"should initialize with a valid block that returns a UIViewController");
+  
+  _viewController = viewControllerBlock();
```

Comment:
Even though we assert, we need to explicitly check if the block is nil before calling it, so that a production app would not crash (showing blank is much better)